drake/systems/DrakeSystem.cpp

```diff
@@ -92,45 +133,129 @@ CascadeSystem::CascadeSystem(const DrakeSystemPtr& _sys1, const DrakeSystemPtr&
   state_frame = make_shared<MultiCoordinateFrame>("CascadeSystemState",initializer_list<CoordinateFramePtr>({continuous_state_frame,discrete_state_frame}));
 }
 
-DrakeSystem::VectorXs CascadeSystem::getX1(const VectorXs &x) const {
-  DrakeSystem::VectorXs x1(sys1->state_frame->getDim());
+VectorXd CascadeSystem::getX1(const VectorXd &x) const {
+  VectorXd x1(sys1->state_frame->getDim());
   x1 << x.head(sys1->continuous_state_frame->getDim()),
         x.segment(sys1->state_frame->getDim(),sys1->discrete_state_frame->getDim());
   return x1;
 }
 
-DrakeSystem::VectorXs CascadeSystem::getX2(const VectorXs &x) const {
-  DrakeSystem::VectorXs x2(sys2->state_frame->getDim());
+VectorXd CascadeSystem::getX2(const VectorXd &x) const {
+  VectorXd x2(sys2->state_frame->getDim());
   x2 << x.segment(sys1->continuous_state_frame->getDim(),sys2->continuous_state_frame->getDim()),
         x.segment(sys1->state_frame->getDim()+sys2->continuous_state_frame->getDim(),sys2->discrete_state_frame->getDim());
   return x2;
 }
 
-DrakeSystem::VectorXs CascadeSystem::dynamics(double t, const DrakeSystem::VectorXs& x, const DrakeSystem::VectorXs& u) const {
+VectorXd CascadeSystem::dynamics(double t, const VectorXd& x, const VectorXd& u) const {
```

Comment:
should put the implementation in a templated function so that we can support the other scalar inputs.